pennylane/_device.py

```diff
@@ -878,6 +878,15 @@ def check_validity(self, queue, observables):
 
             operation_name = o.name
 
+            if getattr(
+                o, "return_type", None
```

Comment:
Is `return_type` ever undefined?

Reply:
Yep, this is a way to check that we have a `MeasurementProcess` in the queue. The only way for this to happen is that we have the return type set to `qml.operation.MidMeasure`.

Reply:
flyby comment: the `return_type` attribute is a remnant of the days where we didn't have the `MeasurementProcess` class, and marked observables that were to be measured by `op.return_type = "expval"`.

We kept it on the `MeasurementProcess` class for backwards compatibility, but we can probably consider better ways of making this developer accessible going forwards!